crates/sui-indexer/src/models_v2/objects.rs

```diff
@@ -161,20 +168,26 @@ impl StoredObject {
                 )))
             }
         };
-        let (name, bcs_name) = if let Some(bcs_name) = self.df_name {
-            let name = bcs::from_bytes(&bcs_name).map_err(|e| {
+        let name = if let Some(field_name) = self.df_name {
+            let name: DynamicFieldName = bcs::from_bytes(&field_name).map_err(|e| {
                 IndexerError::PersistentStorageDataCorruptionError(format!(
                     "object {} has incompatible dynamic field type: df_name. Error: {e}",
                     object_id
                 ))
             })?;
-            Ok::<_, IndexerError>((name, bcs_name))
+            name
         } else {
             return Err(IndexerError::PersistentStorageDataCorruptionError(format!(
                 "object {} has incompatible dynamic field type: empty df_name",
                 object_id
             )));
-        }?;
+        };
+        let layout = move_bytecode_utils::layout::TypeLayoutBuilder::build_with_types(
+            &name.type_,
+            module_cache,
+        )?;
+        let sui_json_value = sui_json::SuiJsonValue::new(name.value.clone())?;
+        let bcs_name = sui_json_value.to_bcs_bytes(&layout)?;
```

Comment:
neat, was worried we'd need to backfill the column to the correct data haha

Reply:
yeah i was worried too, thankfully that's not the case!